### PR TITLE
[CodeCamp2023-476]Add new configuration files for QDTrack algorithm in mmdetection

### PR DESCRIPTION
**Pull request by ZhaoCake**
## Motivation

Add new configuration files for QDTrack algorithm in mmdetection.

## Modification

Two configuration file of qdtrack.

I am sorry for that I confuse the requirement of task, I put new configs into `./configs` yesterday. And now, it has been corrected.

## BC-breaking

As there is no `mot_challenge.py` in `mmdet/configs/_base_/dataset/`, I touched one and made it fit the pure python style 

## Other

- I have finished train and test on A4000-16G, with dataset "MOT17".

- After the modification, it still works well on A4000-16G, with dataset "MOT17".

- It seems unable to pass the checks below. Could anyone tell me the reason? I will modify it.

### CODE REVIEW

**Comment by ZhaoCake** [comment]
I have finished the correction that put configurations into `mmdet/configs`, meanwhile, as there was no mot_challenge.py in `mmdet/configs/_base_/dataset`, I touched one and made it fit pure python style.

**Comment by hhaAndroid** [comment]
@ZhaoCake Please fix the lint. And please provide some logs in the PR message that can demonstrate that the training and testing are normal. Thank you.

File: mmdet/configs/qdtrack/qdtrack_faster_rcnn_r50_fpn_4e_base.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+                          QuasiDenseTrackHead, SingleRoIExtractor,
+                          TrackDataPreprocessor)
+from mmdet.models.losses import (L1Loss, MarginL2Loss,
+                                 MultiPosCrossEntropyLoss, SmoothL1Loss)
+from mmdet.models.task_modules import (CombinedSampler,
+                                       InstanceBalancedPosSampler,
+                                       MaxIoUAssigner, RandomSampler)
+from mmdet.visualization import TrackLocalVisualizer
+
+detector = model
```
Comment (hhaAndroid) [inline mmdet/configs/qdtrack/qdtrack_faster_rcnn_r50_fpn_4e_base.py:28]: 你应该类似 python import 代码写法一样，把 model 显式导入，这样 ci  才能过

Comment (ZhaoCake) [reply mmdet/configs/qdtrack/qdtrack_faster_rcnn_r50_fpn_4e_base.py:29]: 修改后我已经完成了训练和测试。为了 ~节约钱~ 节能环保，低碳训练，我将epoch改为1，将batchsize改为4，训练后生成的config.py以及训练日志、测试日志和scalar.json、权重文件.pth均已存入百度网盘。

[链接:](https://pan.baidu.com/s/1aZoqGt2XT7NnbuxI7bXbcA?pwd=n2fy) 提取码: n2fy

在下方节选日志的尾部内容展示

**Train**
```
2023/07/31 05:01:04 - mmengine - INFO - Epoch(train) [1][1750/1998]  lr: 2.0000e-02  eta: 0:03:29  time: 0.8520  data_time: 0.0257  memory: 11293  grad_norm: 3.0683  loss: 1.0614  loss_rpn_cls: 0.0178  loss_rpn_bbox: 0.0454  loss_cls: 0.1955  acc: 93.1641  loss_bbox: 0.3822  loss_track: 0.2513  loss_track_aux: 0.1692
2023/07/31 05:01:45 - mmengine - INFO - Epoch(train) [1][1800/1998]  lr: 2.0000e-02  eta: 0:02:47  time: 0.8191  data_time: 0.0250  memory: 11293  grad_norm: 3.0205  loss: 1.0113  loss_rpn_cls: 0.0140  loss_rpn_bbox: 0.0411  loss_cls: 0.1875  acc: 90.2832  loss_bbox: 0.3738  loss_track: 0.2310  loss_track_aux: 0.1639
2023/07/31 05:02:27 - mmengine - INFO - Epoch(train) [1][1850/1998]  lr: 2.0000e-02  eta: 0:02:05  time: 0.8476  data_time: 0.0241  memory: 11293  grad_norm: 2.8398  loss: 1.0114  loss_rpn_cls: 0.0162  loss_rpn_bbox: 0.0432  loss_cls: 0.1912  acc: 92.8711  loss_bbox: 0.3711  loss_track: 0.2314  loss_track_aux: 0.1583
2023/07/31 05:03:10 - mmengine - INFO - Epoch(train) [1][1900/1998]  lr: 2.0000e-02  eta: 0:01:22  time: 0.8548  data_time: 0.0247  memory: 11293  grad_norm: 3.1462  loss: 1.0352  loss_rpn_cls: 0.0159  loss_rpn_bbox: 0.0461  loss_cls: 0.1954  acc: 91.0645  loss_bbox: 0.3745  loss_track: 0.2348  loss_track_aux: 0.1685
2023/07/31 05:03:52 - mmengine - INFO - Epoch(train) [1][1950/1998]  lr: 2.0000e-02  eta: 0:00:40  time: 0.8451  data_time: 0.0252  memory: 10932  grad_norm: 3.0269  loss: 1.0050  loss_rpn_cls: 0.0163  loss_rpn_bbox: 0.0443  loss_cls: 0.1890  acc: 93.0664  loss_bbox: 0.3734  loss_track: 0.2279  loss_track_aux: 0.1541
2023/07/31 05:04:33 - mmengine - INFO - Exp name: qdtrack_faster_rcnn_r50_fpn_8xb2-4e_mot17halftrain_test-mot17halfval_20230731_043554
2023/07/31 05:04:33 - mmengine - INFO - Saving checkpoint at 1 epochs
```

**Test**
```
2023/07/31 05:17:15 - mmengine - INFO - Epoch(test) [7950/7956]    eta: 0:00:00  time: 0.0701  data_time: 0.0033  memory: 525  
2023/07/31 05:17:15 - mmengine - INFO - Converting ground truth to coco format...
2023/07/31 05:17:33 - mmengine - INFO - Evaluating bbox...
2023/07/31 05:20:36 - mmengine - INFO - 
+--------------------+-------+--------+--------+-------+-------+-------+
| category           | mAP   | mAP_50 | mAP_75 | mAP_s | mAP_m | mAP_l |
+--------------------+-------+--------+--------+-------+-------+-------+
| pedestrian         | 0.437 | 0.777  | 0.437  | 0.081 | 0.359 | 0.547 |
| person_on_vehicle  | 0.0   | 0.0    | 0.0    | nan   | 0.0   | 0.0   |
| car                | 0.0   | 0.0    | 0.0    | 0.0   | 0.0   | 0.0   |
| bicycle            | 0.0   | 0.0    | 0.0    | nan   | 0.0   | 0.0   |
| motorbike          | 0.0   | 0.0    | 0.0    | nan   | nan   | 0.0   |
| non_mot_vehicle    | 0.0   | 0.0    | 0.0    | nan   | 0.0   | 0.0   |
| static_person      | 0.0   | 0.0    | 0.0    | 0.0   | 0.0   | 0.0   |
| distractor         | 0.0   | 0.0    | 0.0    | nan   | 0.0   | 0.0   |
| occluder           | 0.0   | 0.0    | 0.0    | nan   | 0.0   | 0.0   |
| occluder_on_ground | 0.0   | 0.0    | 0.0    | 0.0   | 0.0   | 0.0   |
| occluder_full      | 0.0   | 0.0    | 0.0    | nan   | 0.0   | 0.0   |
| reflection         | 0.0   | 0.0    | 0.0    | nan   | 0.0   | 0.0   |
| crowd              | nan   | nan    | nan    | nan   | nan   | nan   |
+--------------------+-------+--------+--------+-------+-------+-------+
2023/07/31 05:20:36 - mmengine - INFO - bbox_mAP_copypaste: 0.036 0.065 0.036 0.020 0.033 0.046
2023/07/31 05:20:49 - mmengine - INFO - Evaluating HOTA Metrics...
2023/07/31 05:20:49 - mmengine - INFO - Evaluating CLEAR Metrics...
2023/07/31 05:20:49 - mmengine - INFO - Evaluating Identity Metrics...
2023/07/31 05:20:49 - mmengine - INFO - Epoch(test) [7956/7956]    coco/pedestrian_precision: 0.4370  coco/person_on_vehicle_precision: 0.0000  coco/car_precision: 0.0000  coco/bicycle_precision: 0.0000  coco/motorbike_precision: 0.0000  coco/non_mot_vehicle_precision: 0.0000  coco/static_person_precision: 0.0000  coco/distractor_precision: 0.0000  coco/occluder_precision: 0.0000  coco/occluder_on_ground_precision: 0.0000  coco/occluder_full_precision: 0.0000  coco/reflection_precision: 0.0000  coco/crowd_precision: nan  coco/bbox_mAP: 0.0360  coco/bbox_mAP_50: 0.0650  coco/bbox_mAP_75: 0.0360  coco/bbox_mAP_s: 0.0200  coco/bbox_mAP_m: 0.0330  coco/bbox_mAP_l: 0.0460  motchallenge-metric/HOTA: 0.4950  motchallenge-metric/AssA: 0.4797  motchallenge-metric/DetA: 0.5152  motchallenge-metric/MOTA: 0.6193  motchallenge-metric/MOTP: 0.7967  motchallenge-metric/IDSW: 2730.0000  motchallenge-metric/TP: 108585.0000  motchallenge-metric/FP: 5733.0000  motchallenge-metric/FN: 53079.0000  motchallenge-metric/Frag: 2667.0000  motchallenge-metric/MT: 357.0000  motchallenge-metric/ML: 210.0000  motchallenge-metric/IDF1: 0.6060  motchallenge-metric/IDTP: 83625.0000  motchallenge-metric/IDFN: 78039.0000  motchallenge-metric/IDFP: 30693.0000  motchallenge-metric/IDP: 0.7315  motchallenge-metric/IDR: 0.5173  data_time: 0.0114  time: 0.0877
```

**Comment by OpenMMLab-Assistant-004** [comment]
Hi @ZhaoCake,

We'd like to express our appreciation for your valuable contributions to the mmdetection. Your efforts have significantly aided in enhancing the project's quality.
It is our pleasure to invite you to join our community thorugh Discord_Special Interest Group (SIG) channel. This is a great place to share your experiences, discuss ideas, and connect with other like-minded people. To become a part of the SIG channel, send a message to the moderator, OpenMMLab, briefly introduce yourself and mention your open-source contributions in the #introductions channel. Our team will gladly facilitate your entry. We eagerly await your presence. Please follow this link to join us: ​https://discord.gg/UjgXkPWNqA.

If you're on WeChat, we'd also love for you to join our community there. Just add our assistant using the WeChat ID: openmmlabwx. When sending the friend request, remember to include the remark "mmsig + Github ID".

Thanks again for your awesome contribution, and we're excited to have you as part of our community!

File: mmdet/configs/qdtrack/qdtrack_faster_rcnn_r50_fpn_8xb2-4e_mot17halftrain_test-mot17halfval.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+# evaluator
+val_evaluator = [
+    dict(type=CocoVideoMetric, metric=['bbox'], classwise=True),
+    dict(type=MOTChallengeMetric, metric=['HOTA', 'CLEAR', 'Identity'])
+]
+
+test_evaluator = val_evaluator
+# The fluctuation of HOTA is about +-1.
+randomness = dict(seed=6)
```
Comment (hhaAndroid) [inline mmdet/configs/qdtrack/qdtrack_faster_rcnn_r50_fpn_8xb2-4e_mot17halftrain_test-mot17halfval.py:18]: delete

**Review by hhaAndroid — APPROVED** [review]

File: mmdet/configs/qdtrack/qdtrack_faster_rcnn_r50_fpn_8xb2-4e_mot17halftrain_test-mot17halfval.py (hunk 2 above)
Comment (ZhaoCake) [reply mmdet/configs/qdtrack/qdtrack_faster_rcnn_r50_fpn_8xb2-4e_mot17halftrain_test-mot17halfval.py]: I delete it but it can't pass the test now.

**Review by hhaAndroid — APPROVED** [review]

**Pull request merged**